### PR TITLE
Add cabal-install 3.14.1.0 to the vanilla channel

### PR DESCRIPTION
**Pull request by Kleidukos**
This was also the opportunity to fix some infidelities in the cabal release engineering script for generating ghcup metadata. 

See https://github.com/haskell/cabal/pull/10643.

Pipeline: https://github.com/haskell/ghcup-metadata/actions/runs/12382636335

### CODE REVIEW

**Comment by Kleidukos** [comment]
```
hecate@ifrit on  cabal-3.14.1.0-vanilla [!] via λ 9.6.6 ghcup-metadata ❯ ghcup -v --url-source=file://$(pwd)/ghcup-vanilla-0.0.8.yaml install cabal-install --set 3.14.1.0
[ Warn  ] This is an old-style command for installing GHC. Use 'ghcup install ghc' instead.
[ Debug ] Identified Platform as: Linux Fedora, 41
[ Debug ] using local file: /home/hecate/Contrib/ghcup-metadata/ghcup-vanilla-0.0.8.yaml
[ Debug ] Decoding yaml at: /home/hecate/Contrib/ghcup-metadata/ghcup-vanilla-0.0.8.yaml
[ Debug ] Couldn't decode /home/hecate/Contrib/ghcup-metadata/ghcup-vanilla-0.0.8.yaml as GHCupInfo, trying as SetupInfo: 
[ Debug ] Decoding yaml at: /home/hecate/Contrib/ghcup-metadata/ghcup-vanilla-0.0.8.yaml
[ Error ] [GHCup-00160] JSON decoding failed with: Aeson exception:
[ ...   ] Error in $: key "ghc" not found
[ ...   ] Consider removing /home/hecate/Contrib/ghcup-metadata/ghcup-vanilla-0.0.8.yaml manually.

```

**Comment by Kleidukos** [comment]
I don't know why it took me so long to think of using `ghcup-gen` locally, because this gives me a much better error: 

> Error in $.ghcupDownloads.Cabal['3.14.1.0'].viArch['A_64']['Linux_Alpine']['unknown_versioning']: When parsing the constructor DownloadInfo of type GHCup.Types.DownloadInfo expected Object but got Null.

**Comment by Kleidukos** [comment]
@hasufell Interestingly, the version seems to get lost at some point in CI: https://github.com/haskell/ghcup-metadata/actions/runs/12363842751/job/34505969970#step:5:129

**Comment by Kleidukos** [comment]
I have a hard time understanding why this one fails, is `ghc-9.10` properly set to default? https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506154404#step:5:13111

**Comment by hasufell** [comment]
- https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506154404 suggests the GHC bindist that vanilla picks for RHEL9 is incorrect
- https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506147165 fails expectedly, because upstream does not provide alpine i386 bindists
- https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506149164 same as RHEL9... likely bad GHC bindist selection
- https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506146665 expected failure, armv7 isn't working in newer GHCs
- https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506148278 it looks like cabal binaries for aarch64 alpine are missing 
- https://github.com/haskell/ghcup-metadata/actions/runs/12363903383/job/34506151623 most checks here don't apply for vanilla channel

All in all: vanilla channel isn't very robust. I don't see immediately actionable items here. Those failures are for upstream to fix.

File: ghcup-vanilla-0.0.8.yaml [hunk 1]
```diff
@@ -5408,6 +5407,89 @@ ghcupDownloads:
             unknown_versioning: *cabal-31210-arm64
           Linux_UnknownLinux:
             unknown_versioning: *cabal-31210-arm64
+    3.14.1.0:
+      viTags:
+        - Latest
```
Comment (hasufell) [inline ghcup-vanilla-0.0.8.yaml:5412]: Why is this not recommended?

Comment (Kleidukos) [reply ghcup-vanilla-0.0.8.yaml:5412]: Good question, I'm not sure it makes sense in the vanilla channel until we (upstream) have more feedback on this release?

Comment (hasufell) [reply ghcup-vanilla-0.0.8.yaml:5412]: Most releases just `Latest == Recommended`. At least that's how it's been so far. I think it kinda makes sense, but in the end it's up to the discretion of upstream. 

The main GHCup channel has its own policies.

Comment (Kleidukos) [reply ghcup-vanilla-0.0.8.yaml:5412]: Ok I see

**Pull request merged**